src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,1009 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from argparse import Namespace
+from pydoc import cli
+from kubernetes import client, config, watch, utils
+from binascii import a2b_hex
+from logging import exception
+import os
+import yaml
+import json
+import datetime
+from subprocess import Popen, PIPE, run, STDOUT, call, DEVNULL
+import shutil
+from knack.log import get_logger
+from azure.cli.core import telemetry
+import azext_connectedk8s._constants as consts
+logger = get_logger(__name__)
+# pylint:disable=unused-argument
+# pylint: disable=too-many-locals
+# pylint: disable=too-many-branches
+# pylint: disable=too-many-statements
+# pylint: disable=line-too-long
+
+
+diagnoser_output = []
+
+
+def create_folder_diagnosticlogs(time_stamp):
+
+    global diagnoser_output
+
+    try:
+
+        # Fetching path to current directory to create the arc diagnostic folder
+        home_dir = os.path.expanduser('~')
+        filepath = os.path.join(home_dir, '.azure', 'arc_diagnostic_logs')
+
+        # Creating Diagnostic folder and its subfolder with the given timestamp and cluster name to store all the logs
+        try:
+            os.mkdir(filepath)
+        except FileExistsError:
+            pass
+
+        filepath_with_timestamp = os.path.join(filepath, time_stamp)
+        try:
+            os.mkdir(filepath_with_timestamp)
+        except FileExistsError:
+            pass
+
+        return filepath_with_timestamp, "folder_created"
```

Comment:
This "folder_created" signal string should be stored in constants.py and we should refer using that instead of directly typing it out here.